main.go

```diff
@@ -68,25 +78,25 @@ func main() {
 
 	args, err := parser.Parse()
 	if err != nil {
-		return // message already printed by go-flags
+		return nil // message already printed by go-flags
 	}
 
 	if opts.DisplayVersion {
 		fmt.Printf("thriftrw v%s\n", version.Version)
-		os.Exit(0)
+		return nil
 	}
 
 	if len(args) != 1 {
 		parser.WriteHelp(os.Stdout)
-		os.Exit(1)
+		return errors.New("") // will return non-nil error but print nothing
```

Comment:
This feels a bit hacky. I would rather call `WriteHelp` on a `bytes.Buffer`
and return that as the error.

```
var buff bytes.Buffer
parser.WriteHelp(&buff)
return errors.New(buff.String())
```